atom/browser/native_window_mac.mm

```diff
@@ -95,6 +95,33 @@ - (void)windowDidResignMain:(NSNotification*)notification {
   shell_->NotifyWindowBlur();
 }
 
+- (NSSize)windowWillResize:(NSWindow *)sender toSize:(NSSize)frameSize {
+  NSSize newSize = frameSize;
+  double interiorContentAspectRatio = shell_->GetInteriorContentAspectRatio();
+
+  if (interiorContentAspectRatio > 0.0) {
+    NSRect windowFrame = [sender frame];
+    NSRect contentViewBounds = [[sender contentView] bounds];
```

Comment:
We tried NSBorderlessWindowMask before, but this flag has many side effects like the window not having shadow, and the window can not be key window. So we finally switched to current solution, which is also used by Chromium.